tremor-connectors-azure/src/rest.rs

```diff
@@ -0,0 +1,222 @@
+// Copyright 2024, The Tremor Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use anyhow::{Error, Result};
+use azure_core::headers::Headers;
+use azure_core::Request;
+use azure_core::Response;
+use beef::Cow;
+use bytes::Bytes;
+use simd_json::ObjectHasher;
+use tremor_value::literal;
+use tremor_value::{StaticNode, Value};
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub(crate) struct RequestId(u64);
+
+impl RequestId {
+    pub(crate) fn new(id: u64) -> Self {
+        Self(id)
+    }
+    pub(crate) fn get(self) -> u64 {
+        self.0
+    }
+}
+
+impl std::fmt::Display for RequestId {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.0)
+    }
+}
+
+pub(crate) struct RequestMeta {
+    method: String,
```

Comment:
We have enums for that flying around somewhere :) we can avoid stringly typing the method

Reply:
I kept this similar to the http connector for now as I think it'd be worth another go at this at some point and perhaps consolidate on reqwest - to simplify dependencies for the most part

Reply:
I think we use `SerdeishMethod` (what a wonderful name :D) in `http` to avoid dragging strings aroudn

Reply:
Azure wraps everything http:

```
use azure_core::headers::Headers;
use azure_core::Request;
use azure_core::Response;
```

So i wanted something KISS for experimenting. Ideally, we can get this based on reqwest directly and drop `azure_core` is what I'm thinking...